### DISCUSSION TITLE
Compiling TRIQS using EasyBuild

### DISCUSSION THREAD

**Discussion by TommyHenz** [TRIQS/triqs#865] [Q&A] [answered]
Dear TRIQS community,
I am trying to install TRIQS on my workstation using EasyBuild. I followed the [installations steps](https://triqs.github.io/triqs/latest/install.html#compiling-triqs-from-source-using-easybuild) and I do successfully manage to install, however when I try to load the module using:

`module load TRIQS/3.1.1-foss-2022a`

I get and error saying that:
```
Lmod has detected the following error: The following module(s) are unknown: "TRIQS/3.1.1-foss-2022a"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
    $ module --ignore_cache load "TRIQS/3.1.1-foss-2022a"

Also make sure that all modulefiles written in TCL start with the string #%Module

```

I've tried with the mentioned ways, but I still get this error, I also tried by writing only `module load TRIQS` as well, but nothing happens. Could you let me know how to fix this?

I do have a folder that it created at the path:
./local/easybuild/software/TRIQS/3.1.1-foss-2022a/lib/python3.10/site-packages/triqs

This path has various folders including atom_diag, fit, init.py, etc...

**Comment by the-hampel** [comment, marked as answer]
Hi @TommyHenz,
let's assume your installation worked. Did you add the EB module path to your `$MODULEPATH` environment, i.e. following this guide https://docs.easybuild.io/en/latest/Installation.html#step-3-loading-the-easybuild-module . You can either do this:
```
module use _PREFIX_/modules/all
```
every time you open a new shell, or add this line to your `~/.bashrc` file:
```
export MODULEPATH=_PREFIX_/modules/all:$MODULEPATH
```
where `_PREFIX_` points to your easybuild install directory (maybe `$HOME/easybuild` ? ). If this is successful doing `module available`  should show you all modules available and at which part. Make sure you see your triqs module there. 

Best,
Alex

**Reply by TommyHenz** [reply]
Hello, thanks for your suggestions!
I tried running  `module use  .local/easybuild/modules/all` when I open a shell but I can't load TRIQS.

I tried to see a list of available modules using "module avail" but I can't seem to find TRIQS. In a folder at the following path, I do find TRIQS folder in which there is a .lua file

**.local/easybuild/modules/all/TRIQS**

Does this mean that it is at least correctly installed? It's more of a loading problem right?

**Reply by the-hampel** [reply]
hm, I am not sure. So `echo $MODULEPATH` lists the correct easybuild path and you see other easybuild modules? 

I think I never used the easybuild scripts. Maybe something went wrong and it created some folders, but the installation failed. The install log of easybuild did not show any errors? easybuild should create a log file of your triqs build somewhere, maybe in a tmp folder. Can you maybe share this log file?

**Reply by TommyHenz** [reply]
Wonderful! Running "echo $MODULEPATH" showed me the path that I had. I saw that the path where were I defined my modules (.local/easybuild/modules/all/TRIQS) was on the path from desktop and not from HOME. After correcting them and adjusting it to HOME, I could see all the modules available (which didn't happen before) and I'm able to load TRIQS as well. 

Checked by running `python3 -c "import triqs"` and it works

Thank you so much!
